tests/test_edge_cases.py

```diff
@@ -2161,6 +2184,12 @@ def test_long_int():
     class Model(BaseModel):
         x: int
 
+    # TODO: The next line now raises the following error:
+    #     E       pydantic_core._pydantic_core.ValidationError: 1 validation error for Model
+    #     E       x
+    #     E         Input should be a finite number [type=finite_number,
+    #     input_value='111111111111111111111111...11111111111111111111111', input_type=str]
+    #   Do we need to resolve this? How hard would that be in pydantic_core? Is it worth it?
     assert Model(x='1' * 4_300).x == int('1' * 4_300)
```

Comment:
Question 6 ☝️

Reply:
in `pydantic-core` we use an `i64`, which constrains the max and min values. Since that's massively more performant, and there are very few real world uses for int > `i64:MAX`, the error is correct.

Again, if you really want a bigger int, use the `is-instance` validator and your own validator functions - I'll add this to the migrations guide points now.

Reply:
https://github.com/orgs/pydantic/projects/2